src/types.cpp

```diff
@@ -567,6 +567,9 @@ bool SizedType::FitsInto(const SizedType &t) const
   if (IsStringTy() && t.IsStringTy())
     return GetSize() <= t.GetSize();
 
+  if (IsIntegerTy() && t.IsIntegerTy())
+    return GetSize() <= t.GetSize();
```

Comment:
Should we also check that the signed-ness matches?

e.g. I feel like this shouldn't be allowed:
```
$a = (uint32)1; $a = (int32)-1
```

Reply:
I'm checking the signed-ness before this gets called.

Reply:
I think it needs to be checked in this function, otherwise stuff like this is still possible:
```
$x = (0, (uint32)123); $x = (0, (int32)-123);
```

Reply:
Ah good point. Will also add a test for this case.